File: Practice/src/coding_basic_training/ProductAndSumOfElements.js
function solution(num_list) {
    return num_list.reduce((a, c) => a * c) <
        Math.pow(
            num_list.reduce((a, c) => a + c),
            2
        )
        ? 1
        : 0;
    // Math.pow() 대신 산술연산자도 가능
    // return num_list.reduce((a, c) => a * c) <
    //     num_list.reduce((a, c) => a + c) ** 2
    //     ? 1
    //     : 0;
}

File: Practice/src/coding_basic_training/ProductAndSumOfElements.java
package coding_basic_training;

public class ProductAndSumOfElements {
//	원소들의 곱과 합
	public int solution(int[] num_list) {
		int a = 0, b = 1;
		for (int i = 0; i < num_list.length; i++) {
			a += num_list[i];
			b *= num_list[i];
		}
		return a * a > b ? 1 : 0;
//		Array.Stream을 이용한 방법
//        return (Arrays.stream(numList).reduce((acc, i) -> acc * i).getAsInt() < Math.pow(Arrays.stream(numList).sum(), 2)) ? 1 : 0;

	}
}
